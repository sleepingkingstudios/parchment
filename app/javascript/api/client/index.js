import {
  applyMiddleware,
  selectMiddleware,
} from 'api/middleware/utils';
import {
  valueOrDefault,
} from 'utils/object';

import generateActions from './actions';
import generateHooks from './hooks';
import generateInitialState from './initialState';
import generateReducer from './reducer';
import generateRequest from './request';
import generateSelector from './selector';

const wrapReducer = ({ middleware, reducer }) => applyMiddleware(
  reducer,
  selectMiddleware(middleware, 'handleAction'),
);

const buildClient = (options) => {
  const {
    namespace,
    url,
  } = options;
  const data = valueOrDefault(options.data, {});
  const method = valueOrDefault(options.method, 'GET');
  const middleware = valueOrDefault(options.middleware, []);

  const actions = generateActions({ namespace });
  const initialState = generateInitialState({ data });
  const performRequest = generateRequest({
    actions,
    method,
    middleware,
    namespace,
    url,
  });
  const hooks = generateHooks({ performRequest });
  const reducer = wrapReducer({
    middleware,
    reducer: generateReducer({ actions, initialState }),
  });
  const selector = generateSelector({ namespace });

  return {
    actions,
    hooks,
    method,
    middleware,
    namespace,
    performRequest,
    reducer,
    selector,
    url,
  };
};

export default buildClient;
