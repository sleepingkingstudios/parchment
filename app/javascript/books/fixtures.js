import { load } from '../fixtures';

/* eslint-disable-next-line import/prefer-default-export */
export const booksData = load('books');
