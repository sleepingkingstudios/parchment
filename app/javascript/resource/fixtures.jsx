import React from 'react';

export const Block = () => (<div />);
export const Form = () => (<div />);
export const Table = () => (<div />);

export const CreatePage = () => (<div />);
export const IndexPage = () => (<div />);
export const LandingPage = () => (<div />);
export const ShowPage = () => (<div />);
export const UpdatePage = () => (<div />);

export const fixtures = {
  create: {
    component: CreatePage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/create',
    reducer: ((state = { widget: {} }) => state),
  },
  index: {
    component: IndexPage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/',
    reducer: ((state = { widgets: [] }) => state),
  },
  show: {
    component: ShowPage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/:id',
    reducer: ((state = { widget: {} }) => state),
  },
  update: {
    component: UpdatePage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/:id/update',
    reducer: ((state = { widget: {} }) => state),
  },
};
