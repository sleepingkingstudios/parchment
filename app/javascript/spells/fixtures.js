/* eslint-disable-next-line import/prefer-default-export */
export const spellsData = [
  {
    id: 'e17bca01-24d3-48d3-a308-099dd6dca373',
    level: 0,
    name: 'Embers',
    school: 'Evocation',
  },
  {
    id: 'e3641970-fc98-4674-b873-5532d89d9f26',
    level: 3,
    name: "Dragon's Breath",
    school: 'Evocation',
  },
  {
    id: 'a74d8c21-607d-4a3f-8cfd-2bbb48ec196a',
    level: 9,
    name: 'Wrath of the Titans',
    school: 'Evocation',
  },
];
