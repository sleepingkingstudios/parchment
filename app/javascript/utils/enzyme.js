import { valueOrDefault } from './object';
import { safeCapitalize } from './string';

export const extractInjectedComponent = (component) => {
  const injected = component.shallow().find('ComponentWithInjectedProps');

  return injected.exists() ? injected : component;
};

export const toHaveFormControl = expect => (received, expectedControl, withProps) => {
  const formControl = received.find(expectedControl);
  const expectedProps = valueOrDefault(withProps, {});

  expect(formControl).toExist();
  expect(formControl).toHaveProp(expectedProps);

  return { pass: true, message: () => '' };
};

export const toHaveFormGroup = expect => (received, expectedGroup, fn) => {
  const formGroup = received.find(expectedGroup);

  expect(formGroup).toExist();

  fn(extractInjectedComponent(formGroup).shallow());

  return { pass: true, message: () => '' };
};

export const toHaveFormInput = expect => (received, props) => {
  const {
    form,
    inputType,
    prop,
  } = props;

  expect(received).toHaveProp({ prop });

  const formInput = received.find(`${safeCapitalize(prop)}Input`);

  expect(formInput).toExist();
  expect(formInput).toHaveProp({ form });

  const shallowInput = formInput.shallow();

  expect(shallowInput).toExist();
  expect(shallowInput).toHaveDisplayName(valueOrDefault(inputType, 'FormInput'));

  return { pass: true, message: () => '' };
};
