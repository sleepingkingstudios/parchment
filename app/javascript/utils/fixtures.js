/* eslint-disable-next-line import/prefer-default-export */
export const trimMultilineLiteral = (text) => {
  if (!text) { return ''; }

  const paragraphs = text.split('\n\n');

  return paragraphs
    .map((paragraph) => {
      const lines = paragraph.split('\n');

      return lines.map(line => line.trim()).join(' ');
    })
    .join('\n\n');
};
