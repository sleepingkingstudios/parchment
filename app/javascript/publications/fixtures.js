import { load } from '../fixtures';

const isOfficial = (publication) => {
  const { playtest, publisherName } = publication;

  return !playtest && publisherName === 'Wizards of the Coast';
};

const rawData = load('publications');

/* eslint-disable-next-line import/prefer-default-export */
export const publicationsData = rawData
  .map(publication => Object.assign(
    {},
    publication,
    {
      official: isOfficial(publication),
    },
  ));
