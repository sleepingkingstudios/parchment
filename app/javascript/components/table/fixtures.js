import { humanizeList } from '../../utils/array';

export const formatCrewCapacity = (item) => {
  const count = item.crewCapacity;

  if (count === 1) { return '1 Kerbal'; }

  return `${count} Kerbals`;
};

export const formatExperiments = (item) => {
  humanizeList(item.experiments, { default: 'None' });
};

export const columns = [
  {
    label: 'Name',
    prop: 'name',
  },
  {
    label: 'Crew Capacity',
    prop: 'crewCapacity',
    value: formatCrewCapacity,
  },
  {
    label: 'Experiments',
    prop: 'experiments',
  },
];

export const rocketsData = [
  {
    id: '83f5e9e8-92df-4203-bc2e-4381370c2186',
    name: 'Apprentice',
    crewCapacity: 1,
    experiments: [
      'Crew Report',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
    ],
  },
  {
    id: '4bd7454c-3b15-4047-8d57-dfa831dac58e',
    name: 'Sorcerer',
    crewCapacity: 2,
    experiments: [
      'Crew Report',
      'Materials Study',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
    ],
  },
  {
    id: '2fead083-d9ea-4f04-949e-d8c00e800cbb',
    name: 'Warlock',
    crewCapacity: 3,
    experiments: [
      'Crew Report',
      'Materials Study',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
      'Gravity Scan',
    ],
  },
];
