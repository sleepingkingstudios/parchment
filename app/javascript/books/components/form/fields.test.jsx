import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';

import {
  CancelButton,
  PublicationDateField,
  PublisherNameField,
  SlugField,
  SubmitButton,
  TitleField,
} from './fields';
import { booksData } from '../../fixtures';
import { INITIALIZED } from '../../../api/status';

describe('<BookForm /> fields', () => {
  const onChangeAction = jest.fn(
    ({ propName, value }) => ({ payload: { propName, value } }),
  );
  const onSubmitAction = jest.fn(() => ({ payload: { ok: true } }));
  const book = booksData[0];
  const form = {
    data: { book },
    path: ['book'],
    onChangeAction,
    onSubmitAction,
  };

  describe('<CancelButton />', () => {
    const children = 'Do Nothing';
    const defaultProps = {
      children,
      form,
      isUpdate: false,
    };

    it('should set the display name', () => {
      expect(CancelButton.displayName).toEqual('CancelButton');
    });

    it('should render a form group', () => {
      const rendered = shallow(<CancelButton {...defaultProps} />);

      expect(rendered).toHaveDisplayName('FormGroup');
    });

    it('should match the snapshot', () => {
      const rendered = mount(<CancelButton {...defaultProps} />, { wrappingComponent: Router });

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('<PublicationDateField />', () => {
    const defaultProps = { form };

    it('should set the display name', () => {
      expect(PublicationDateField.displayName).toEqual('PublicationDateField');
    });

    it('should render a form field', () => {
      const rendered = shallow(<PublicationDateField {...defaultProps} />);

      expect(rendered).toHaveDisplayName('FormField');
    });

    it('should match the snapshot', () => {
      const rendered = mount(<PublicationDateField {...defaultProps} />);

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('<PublisherNameField />', () => {
    const defaultProps = { form };

    it('should set the display name', () => {
      expect(PublisherNameField.displayName).toEqual('PublisherNameField');
    });

    it('should render a form field', () => {
      const rendered = shallow(<PublisherNameField {...defaultProps} />);

      expect(rendered).toHaveDisplayName('FormField');
    });

    it('should match the snapshot', () => {
      const rendered = mount(<PublisherNameField {...defaultProps} />);

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('<SlugField />', () => {
    const defaultProps = { form };

    it('should set the display name', () => {
      expect(SlugField.displayName).toEqual('SlugField');
    });

    it('should render a form field', () => {
      const rendered = shallow(<SlugField {...defaultProps} />);

      expect(rendered).toHaveDisplayName('FormField');
    });

    it('should match the snapshot', () => {
      const rendered = mount(<SlugField {...defaultProps} />);

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('<SubmitButton />', () => {
    const children = 'Do Something';
    const defaultProps = { children, form, status: INITIALIZED };

    it('should set the display name', () => {
      expect(SubmitButton.displayName).toEqual('SubmitButton');
    });

    it('should render a form group', () => {
      const rendered = shallow(<SubmitButton {...defaultProps} />);

      expect(rendered).toHaveDisplayName('FormGroup');
    });

    it('should match the snapshot', () => {
      const rendered = mount(<SubmitButton {...defaultProps} />);

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('<TitleField />', () => {
    const defaultProps = { form };

    it('should set the display name', () => {
      expect(TitleField.displayName).toEqual('TitleField');
    });

    it('should render a form field', () => {
      const rendered = shallow(<TitleField {...defaultProps} />);

      expect(rendered).toHaveDisplayName('FormField');
    });

    it('should match the snapshot', () => {
      const rendered = mount(<TitleField {...defaultProps} />);

      expect(rendered).toMatchSnapshot();
    });
  });
});
