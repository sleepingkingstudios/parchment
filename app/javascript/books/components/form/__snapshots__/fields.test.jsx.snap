// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookForm /> fields <CancelButton /> should match the snapshot 1`] = `
<CancelButton
  form={
    Object {
      "data": Object {
        "book": Object {
          "abbreviation": "sff",
          "id": "38a6666f-d3eb-4853-bd0a-add4d8ec369a",
          "playtest": true,
          "publicationDate": "2013-08-15",
          "publisherName": "Spelljammer Publishing",
          "slug": "spectres-flumph-fortress",
          "title": "Spectres of Flumph Fortress",
        },
      },
      "onChangeAction": [MockFunction],
      "onSubmitAction": [MockFunction],
      "path": Array [
        "book",
      ],
    }
  }
  isUpdate={false}
>
  <FormGroup
    className=""
    colWidth={null}
  >
    <div
      className="form-group col"
    >
      <BookFormCancelButton
        form={
          Object {
            "data": Object {
              "book": Object {
                "abbreviation": "sff",
                "id": "38a6666f-d3eb-4853-bd0a-add4d8ec369a",
                "playtest": true,
                "publicationDate": "2013-08-15",
                "publisherName": "Spelljammer Publishing",
                "slug": "spectres-flumph-fortress",
                "title": "Spectres of Flumph Fortress",
              },
            },
            "onChangeAction": [MockFunction],
            "onSubmitAction": [MockFunction],
            "path": Array [
              "book",
            ],
          }
        }
        isUpdate={false}
      >
        <FormCancelButton
          baseUrl={null}
          form={
            Object {
              "data": Object {
                "book": Object {
                  "abbreviation": "sff",
                  "id": "38a6666f-d3eb-4853-bd0a-add4d8ec369a",
                  "playtest": true,
                  "publicationDate": "2013-08-15",
                  "publisherName": "Spelljammer Publishing",
                  "slug": "spectres-flumph-fortress",
                  "title": "Spectres of Flumph Fortress",
                },
              },
              "onChangeAction": [MockFunction],
              "onSubmitAction": [MockFunction],
              "path": Array [
                "book",
              ],
            }
          }
          isUpdate={false}
          propName="slug"
          resourceName="Book"
        >
          <LinkButton
            block={true}
            buttonSize={null}
            buttonStyle="secondary"
            className={null}
            link={false}
            outline={true}
            url="/books"
          >
            <Link
              className="btn btn-outline-secondary btn-block"
              to="/books"
            >
              <a
                className="btn btn-outline-secondary btn-block"
                href="/books"
                onClick={[Function]}
              >
                Cancel
              </a>
            </Link>
          </LinkButton>
        </FormCancelButton>
      </BookFormCancelButton>
    </div>
  </FormGroup>
</CancelButton>
`;

exports[`<BookForm /> fields <PublicationDateField /> should match the snapshot 1`] = `
<PublicationDateField
  form={
    Object {
      "data": Object {
        "book": Object {
          "abbreviation": "sff",
          "id": "38a6666f-d3eb-4853-bd0a-add4d8ec369a",
          "playtest": true,
          "publicationDate": "2013-08-15",
          "publisherName": "Spelljammer Publishing",
          "slug": "spectres-flumph-fortress",
          "title": "Spectres of Flumph Fortress",
        },
      },
      "onChangeAction": [MockFunction],
      "onSubmitAction": [MockFunction],
      "path": Array [
        "book",
      ],
    }
  }
>
  <FormField
    className="book-publication-date-field"
    colWidth={null}
    inputId={null}
    label={null}
    path={
      Array [
        "book",
      ]
    }
    prop="publicationDate"
  >
    <FormGroup
      className="form-field book-publication-date-field"
      colWidth={null}
    >
      <div
        className="form-group col form-field book-publication-date-field"
      >
        <label
          htmlFor="book-publication-date-input"
        >
          Publication Date
        </label>
        <PublicationDateInput
          form={
            Object {
              "data": Object {
                "book": Object {
                  "abbreviation": "sff",
                  "id": "38a6666f-d3eb-4853-bd0a-add4d8ec369a",
                  "playtest": true,
                  "publicationDate": "2013-08-15",
                  "publisherName": "Spelljammer Publishing",
                  "slug": "spectres-flumph-fortress",
                  "title": "Spectres of Flumph Fortress",
                },
              },
              "onChangeAction": [MockFunction],
              "onSubmitAction": [MockFunction],
              "path": Array [
                "book",
              ],
            }
          }
        >
          <FormInput
            className={null}
            id="book-publication-date-input"
            onChange={[Function]}
            placeholder={null}
            type="text"
            validStatus={null}
            value="2013-08-15"
          >
            <input
              className="form-control"
              id="book-publication-date-input"
              onChange={[Function]}
              placeholder={null}
              type="text"
              value="2013-08-15"
            />
          </FormInput>
        </PublicationDateInput>
      </div>
    </FormGroup>
  </FormField>
</PublicationDateField>
`;

exports[`<BookForm /> fields <PublisherNameField /> should match the snapshot 1`] = `
<PublisherNameField
  form={
    Object {
      "data": Object {
        "book": Object {
          "abbreviation": "sff",
          "id": "38a6666f-d3eb-4853-bd0a-add4d8ec369a",
          "playtest": true,
          "publicationDate": "2013-08-15",
          "publisherName": "Spelljammer Publishing",
          "slug": "spectres-flumph-fortress",
          "title": "Spectres of Flumph Fortress",
        },
      },
      "onChangeAction": [MockFunction],
      "onSubmitAction": [MockFunction],
      "path": Array [
        "book",
      ],
    }
  }
>
  <FormField
    className="book-publisher-name-field"
    colWidth={null}
    inputId={null}
    label={null}
    path={
      Array [
        "book",
      ]
    }
    prop="publisherName"
  >
    <FormGroup
      className="form-field book-publisher-name-field"
      colWidth={null}
    >
      <div
        className="form-group col form-field book-publisher-name-field"
      >
        <label
          htmlFor="book-publisher-name-input"
        >
          Publisher Name
        </label>
        <PublisherNameInput
          form={
            Object {
              "data": Object {
                "book": Object {
                  "abbreviation": "sff",
                  "id": "38a6666f-d3eb-4853-bd0a-add4d8ec369a",
                  "playtest": true,
                  "publicationDate": "2013-08-15",
                  "publisherName": "Spelljammer Publishing",
                  "slug": "spectres-flumph-fortress",
                  "title": "Spectres of Flumph Fortress",
                },
              },
              "onChangeAction": [MockFunction],
              "onSubmitAction": [MockFunction],
              "path": Array [
                "book",
              ],
            }
          }
        >
          <FormInput
            className={null}
            id="book-publisher-name-input"
            onChange={[Function]}
            placeholder={null}
            type="text"
            validStatus={null}
            value="Spelljammer Publishing"
          >
            <input
              className="form-control"
              id="book-publisher-name-input"
              onChange={[Function]}
              placeholder={null}
              type="text"
              value="Spelljammer Publishing"
            />
          </FormInput>
        </PublisherNameInput>
      </div>
    </FormGroup>
  </FormField>
</PublisherNameField>
`;

exports[`<BookForm /> fields <SlugField /> should match the snapshot 1`] = `
<SlugField
  form={
    Object {
      "data": Object {
        "book": Object {
          "abbreviation": "sff",
          "id": "38a6666f-d3eb-4853-bd0a-add4d8ec369a",
          "playtest": true,
          "publicationDate": "2013-08-15",
          "publisherName": "Spelljammer Publishing",
          "slug": "spectres-flumph-fortress",
          "title": "Spectres of Flumph Fortress",
        },
      },
      "onChangeAction": [MockFunction],
      "onSubmitAction": [MockFunction],
      "path": Array [
        "book",
      ],
    }
  }
>
  <FormField
    className="book-slug-field"
    colWidth={null}
    inputId={null}
    label={null}
    path={
      Array [
        "book",
      ]
    }
    prop="slug"
  >
    <FormGroup
      className="form-field book-slug-field"
      colWidth={null}
    >
      <div
        className="form-group col form-field book-slug-field"
      >
        <label
          htmlFor="book-slug-input"
        >
          Slug
        </label>
        <SlugInput
          form={
            Object {
              "data": Object {
                "book": Object {
                  "abbreviation": "sff",
                  "id": "38a6666f-d3eb-4853-bd0a-add4d8ec369a",
                  "playtest": true,
                  "publicationDate": "2013-08-15",
                  "publisherName": "Spelljammer Publishing",
                  "slug": "spectres-flumph-fortress",
                  "title": "Spectres of Flumph Fortress",
                },
              },
              "onChangeAction": [MockFunction],
              "onSubmitAction": [MockFunction],
              "path": Array [
                "book",
              ],
            }
          }
        >
          <FormInput
            className={null}
            id="book-slug-input"
            onChange={[Function]}
            placeholder="spectres-flumph-fortress"
            type="text"
            validStatus={null}
            value="spectres-flumph-fortress"
          >
            <input
              className="form-control"
              id="book-slug-input"
              onChange={[Function]}
              placeholder="spectres-flumph-fortress"
              type="text"
              value="spectres-flumph-fortress"
            />
          </FormInput>
        </SlugInput>
      </div>
    </FormGroup>
  </FormField>
</SlugField>
`;

exports[`<BookForm /> fields <SubmitButton /> should match the snapshot 1`] = `
<SubmitButton
  disabled={false}
  form={
    Object {
      "data": Object {
        "book": Object {
          "abbreviation": "sff",
          "id": "38a6666f-d3eb-4853-bd0a-add4d8ec369a",
          "playtest": true,
          "publicationDate": "2013-08-15",
          "publisherName": "Spelljammer Publishing",
          "slug": "spectres-flumph-fortress",
          "title": "Spectres of Flumph Fortress",
        },
      },
      "onChangeAction": [MockFunction],
      "onSubmitAction": [MockFunction],
      "path": Array [
        "book",
      ],
    }
  }
  status="initialized"
>
  <FormGroup
    className=""
    colWidth={null}
  >
    <div
      className="form-group col"
    >
      <BookFormSubmitButton
        disabled={false}
        form={
          Object {
            "data": Object {
              "book": Object {
                "abbreviation": "sff",
                "id": "38a6666f-d3eb-4853-bd0a-add4d8ec369a",
                "playtest": true,
                "publicationDate": "2013-08-15",
                "publisherName": "Spelljammer Publishing",
                "slug": "spectres-flumph-fortress",
                "title": "Spectres of Flumph Fortress",
              },
            },
            "onChangeAction": [MockFunction],
            "onSubmitAction": [MockFunction],
            "path": Array [
              "book",
            ],
          }
        }
        status="initialized"
      >
        <FormSubmitButton
          actionName="Create"
          form={
            Object {
              "data": Object {
                "book": Object {
                  "abbreviation": "sff",
                  "id": "38a6666f-d3eb-4853-bd0a-add4d8ec369a",
                  "playtest": true,
                  "publicationDate": "2013-08-15",
                  "publisherName": "Spelljammer Publishing",
                  "slug": "spectres-flumph-fortress",
                  "title": "Spectres of Flumph Fortress",
                },
              },
              "onChangeAction": [MockFunction],
              "onSubmitAction": [MockFunction],
              "path": Array [
                "book",
              ],
            }
          }
          resourceName="Book"
        >
          <Button
            block={true}
            buttonSize="md"
            buttonStyle="primary"
            className="form-submit book-form-submit"
            disabled={false}
            link={false}
            onClick={[Function]}
            outline={true}
          >
            <button
              className="btn btn-outline-primary btn-block form-submit book-form-submit"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Create Book
            </button>
          </Button>
        </FormSubmitButton>
      </BookFormSubmitButton>
    </div>
  </FormGroup>
</SubmitButton>
`;

exports[`<BookForm /> fields <TitleField /> should match the snapshot 1`] = `
<TitleField
  form={
    Object {
      "data": Object {
        "book": Object {
          "abbreviation": "sff",
          "id": "38a6666f-d3eb-4853-bd0a-add4d8ec369a",
          "playtest": true,
          "publicationDate": "2013-08-15",
          "publisherName": "Spelljammer Publishing",
          "slug": "spectres-flumph-fortress",
          "title": "Spectres of Flumph Fortress",
        },
      },
      "onChangeAction": [MockFunction],
      "onSubmitAction": [MockFunction],
      "path": Array [
        "book",
      ],
    }
  }
>
  <FormField
    className="book-title-field"
    colWidth={null}
    inputId={null}
    label={null}
    path={
      Array [
        "book",
      ]
    }
    prop="title"
  >
    <FormGroup
      className="form-field book-title-field"
      colWidth={null}
    >
      <div
        className="form-group col form-field book-title-field"
      >
        <label
          htmlFor="book-title-input"
        >
          Title
        </label>
        <TitleInput
          form={
            Object {
              "data": Object {
                "book": Object {
                  "abbreviation": "sff",
                  "id": "38a6666f-d3eb-4853-bd0a-add4d8ec369a",
                  "playtest": true,
                  "publicationDate": "2013-08-15",
                  "publisherName": "Spelljammer Publishing",
                  "slug": "spectres-flumph-fortress",
                  "title": "Spectres of Flumph Fortress",
                },
              },
              "onChangeAction": [MockFunction],
              "onSubmitAction": [MockFunction],
              "path": Array [
                "book",
              ],
            }
          }
        >
          <FormInput
            className={null}
            id="book-title-input"
            onChange={[Function]}
            placeholder={null}
            type="text"
            validStatus={null}
            value="Spectres of Flumph Fortress"
          >
            <input
              className="form-control"
              id="book-title-input"
              onChange={[Function]}
              placeholder={null}
              type="text"
              value="Spectres of Flumph Fortress"
            />
          </FormInput>
        </TitleInput>
      </div>
    </FormGroup>
  </FormField>
</TitleField>
`;
